solver/src/settlement/settlement_encoder.rs

```diff
@@ -262,6 +262,10 @@ impl SettlementEncoder {
     // both or more than one token has a different clearing prices (a single token difference is scaled)
     pub fn merge(mut self, mut other: Self) -> Result<Self> {
         let scaling_factor = self.price_scaling_factor(&other);
+        // Make sure we always scale prices up to avoid precision issues
+        if scaling_factor < BigRational::from_integer(1.into()) {
```

Comment:
I think that can use BugRational::one() here. Unless you're trying to avoid an ugly import.